docs/extensibility/resources.md

```diff
@@ -0,0 +1,224 @@
+
+The Config Map needs to have several fields created that define how the resources are handled.
+
+All sections can be provided as either JSON or YAML.
+
+## Config Map sections
+
+### resource section
+
+The `resource` section is required and contains basic information about the resource. For example, `kind` and api details.
+
+#### Example
+
+```json
+{
+  "kind": "MyResource",
+  "group": "networking.istio.io",
+  "version": "v1alpha3"
+}
+```
+
+### navigation section
+
+The `navigation` section is required and defines this resource availability in the menu and the generated url. The name of the menu entry as well as category that it's going to be placed in are read from the translation section entries `name` and `category` respectively.
+
+#### Item paremeters
+
+- **path** - path for this resource used in the url.
+- **scope** - either `namespace` or `cluster`.
+
+#### Example
+
+```json
+{
+  "path": "my-resources",
+  "scope": "namespace"
+}
+```
+
+### form section
+
+The `form` section modifies the original schema (provided in the schema section) for use with [ui-schema](https://ui-schema.bemit.codes/) to generate the create/edit form for the resource. It contains a list of objects that define which fields must be included in the final form. All given fields are placed in the advanced form by default. It's possible to add a field to simple form by providing the `simple: true` flag, or removing it from advanced form by providing the `advanced: false` flag.
+
+#### Item parameters
+
+- **path** - _[required]_ path to the property that must be displayed in the form. In case of an array the array index is omitted. For example, if `spec.items` is an array and you want to display `name` for each items, the path is `spec.items.name`).
+- **widget** - optional widget to render the field. If no widget is provided a default handler is used depending on the data type provided in the schema. For more information about the available widgets, see [Form widgets](form-widgets.md).
+- **simple** - to display in the simple form. By default it is false.
+- **advanced** - to display in the advanced form. By default it is true.
+
+### Example
+
+```json
+[
+  { "path": "spec.priority", "simple": true },
+  { "path": "spec.items.name" },
+  { "path": "spec.items.service.url" },
+  { "path": "spec.items.service.port" }
+]
+```
+
+### list section
+
+The `list` section defines extra columns available in the list. The format is similar to the form section, however each entry consists only of two values:
+
+#### Item parameters
+
+- **path** - _[required]_ contains the path to the data used for the column.
+- **widget** - optional widget used to render the field. By default the value is displayed verbatim. For more information about the available widgets, see [Display widgets](display-widgets.md).
```

Comment:
```suggestion
- **widget** - optional widget used to render the field. By default the value is displayed verbatim. For more information about the available widgets, see [Display widgets](display-widgets.md).
```
What type of field this parameter renders? If it's not specific use `a field`. If it's specific please provide more details (e.g. object field, array field,...)